### PR TITLE
CON-3593 report warnings and errors from #ldap_sync_policy

### PR DESCRIPTION
**Pull request by apotterri**
Exit non-zero if policy generation fails.

### CODE REVIEW

File: lib/conjur/command/ldapsync.rb [hunk 1]
```diff
@@ -4,14 +4,16 @@ class Conjur::Command::LDAPSync < Conjur::Command
 
   LIST_FORMATS = %w(pretty json)
 
-  def self.find_job_by_id args
```
Comment (apotterri) [inline lib/conjur/command/ldapsync.rb]: Missed this one when I cleaned up other job-related functionality.

File: lib/conjur/command/ldapsync.rb [hunk 2]
```diff
@@ -4,14 +4,16 @@ class Conjur::Command::LDAPSync < Conjur::Command
...
 
-    if (job = api.ldap_sync_jobs.find{|j| j.id == job_id})
-      job
-    else
-      exit_now! "No job found with ID '#{job_id}'"
-    end
+  def self.show_messages(resp)
+    msgs = resp['events'].collect do |e|
+      next unless e['severity'] == 'warn' || e['severity'] == 'error'
+      "\n" + e['severity'].upcase + ": " + e['message']
```
Comment (jvanderhoof) [inline lib/conjur/command/ldapsync.rb]: Will starting each line with a new line character cause an extra line to be added to the start of this output?  Would a `join("\n")` work better at the end of the `msgs` array?

Comment (jvanderhoof) [inline lib/conjur/command/ldapsync.rb]: Any reason you went with string concatenation over interpolation? What do you think about sticking with interpolation for the sake of project convention?

Comment (apotterri) [inline lib/conjur/command/ldapsync.rb]: I didn't think it was extra. It's the difference between
```
$ conjur ldap-sync policy show

WARN: something not so bad happened

WARN: another not so bad thing happend

ERROR: something really bad happened

error: Failed creating the policy.
```

and

```
$ conjur ldap-sync policy show
WARN: something not so bad happened
WARN: another not so bad thing happend
ERROR: something really bad happened
error: Failed creating the policy.
```

I thought the former was more readable, especially as the messages got longer.

Comment (apotterri) [inline lib/conjur/command/ldapsync.rb]: Should be

```
  if e['severity']....
```

but otherwise I like that. I'll make that change.

Comment (apotterri) [inline lib/conjur/command/ldapsync.rb]: And, re interpolation versus concatenation, it was a holdover from a time when that expression was more complicated. I agree that, in its current form, it'll be better as interpolation.

Comment (jvanderhoof) [inline lib/conjur/command/ldapsync.rb]: I agree with you on the readability front with the new line characters.  Hard to make anything look good in the terminal, but the extra space really helps.

Comment (jvanderhoof) [inline lib/conjur/command/ldapsync.rb]: I've run Rubocop at my last two companies, and I fear it's made me more hyper aware of code style. :)

**Pull request merged**